### ISSUE TITLE
rtx executable is super slow on my system.

### ISSUE THREAD

**Issue by cocoonkid** [jdx/rtx#918]
Super slow :)

I just removed .asdf and asdf traces from my zshrc and installed rtx via the AUR.

I use rtx already on my other system so did not expect this difference in behaviour.

Running Arch Linux latest kernel.
rtx 2023.9.2 linux-x64 (155dadb 2023-09-25)

I tried:

- rtx implode.
- deleting .local/share/rtx manually
- tried to find more traces of .asdf that might cause this but nothing.

[TRACE] (28) reqwest::blocking::wait: [/cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.20/src/blocking/wait.rs:42] 

is just repeating for quite some time until it times out and actually does something. 

love rtx btw :-)

Here is my debug trace so far.
```
╭─cocoonkid@arch in ~ via  v20.7.0 via  v3.11.5 took 457ms
[🔴] × rtx list
[WARN] rtx::toolset::tool_version_list: failed to resolve tool version: timed out: timed out waiting on channel
assh   2.15.0          ~/.tool-versions 2.15.0
direnv 2.32.3          ~/.tool-versions 2.32.3
go     1.21.1          ~/.tool-versions 1.21.1
node   20.7.0          ~/.tool-versions 20.7.0
php    7.4.33 (missing) ~/.tool-versions 7.4.33
poetry 1.5.1           ~/.tool-versions 1.5.1
python 3.11.5          ~/.tool-versions 3.11.5
rust   1.72.1          ~/.tool-versions 1.72.1
Time: 0h:00m:10s

 ╭─cocoonkid@arch in ~ via  v20.7.0 via  v3.11.5 took 10s
 ╰─λ RTX_TRACE=1 rtx list
[TRACE] (1) rtx::config: [src/config/mod.rs:263] settings does not exist "/home/cocoonkid/.config/rtx/config.toml"
[TRACE] (1) rtx::plugins::rtx_plugin_toml: [src/plugins/rtx_plugin_toml.rs:28] parsing: /home/cocoonkid/.local/share/rtx/plugins/poetry/rtx.plugin.toml
[TRACE] (1) rtx::file: [src/file.rs:66] cat /home/cocoonkid/.local/share/rtx/plugins/poetry/rtx.plugin.toml
[TRACE] (1) rtx::config::config_file::tool_versions: [src/config/config_file/tool_versions.rs:56] parsing tool-versions: /home/cocoonkid/.tool-versions
[TRACE] (1) rtx::file: [src/file.rs:66] cat /home/cocoonkid/.tool-versions
[TRACE] (1) rtx::config: [src/config/mod.rs:68] Settings: Settings {
    experimental: false,
    missing_runtime_behavior: Warn,
    always_keep_download: false,
    always_keep_install: false,
    legacy_version_file: true,
    legacy_version_file_disable_tools: {},
    plugin_autoupdate_last_check_duration: 604800s,
    trusted_config_paths: {},
    verbose: true,
    asdf_compat: false,
    jobs: 4,
    shorthands_file: None,
    disable_default_shorthands: false,
    disable_tools: {},
    log_level: Trace,
    raw: false,
    yes: false,
}
[DEBUG] (1) rtx::config: [src/config/mod.rs:111] Files: ~/.tool-versions
Installed Plugins: assh, direnv, php, poetry, rust
[DEBUG] (1) rtx::cli: [src/cli/mod.rs:191] rtx list
[TRACE] (6) rtx::cache: [src/cache.rs:81] reading ~/.cache/rtx/php/remote_versions.msgpack.z
[TRACE] (28) reqwest::blocking::wait: [/cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.20/src/blocking/wait.rs:49] (ThreadId(28)) park without timeout
[TRACE] (29) reqwest::blocking::client: [/cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.20/src/blocking/client.rs:1047] (ThreadId(29)) start runtime::block_on
[DEBUG] (28) rtx::http: [src/http.rs:21] GET https://joschi.github.io/java-metadata/metadata/all.json
[TRACE] (28) reqwest::blocking::wait: [/cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.20/src/blocking/wait.rs:16] wait at most 30s
[TRACE] (28) reqwest::blocking::wait: [/cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.20/src/blocking/wait.rs:42] (ThreadId(28)) park timeout 29.999998167s
[DEBUG] (29) reqwest::connect: [/cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.20/src/connect.rs:429] starting new connection: https://joschi.github.io/
^C[DEBUG] (2) rtx: [src/main.rs:96] Ctrl-C pressed, exiting...
```

```
rtx doctor:

rtx version:
  2023.9.2 linux-x64 (155dadb 2023-09-25)

build:
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS, OPENSSL, SELF_UPDATE
  Built: Mon, 25 Sep 2023 17:09:06 +0000
  Rust Version: rustc 1.72.0 (5680fa18f 2023-08-23)
  Profile: release

shell:
  zsh
  zsh 5.9 (x86_64-pc-linux-gnu)

rtx data directory:
  /home/cocoonkid/.local/share/rtx

rtx environment variables:
  RTX_SHELL=zsh

settings:
  {"always_keep_download": "false", "always_keep_install": "false", "asdf_compat": "false", "disable_default_shorthands": "false", "disable_tools": "[]", "experimental": "false", "jobs": "4", "legacy_version_file": "true", "legacy_version_file_disable_tools": "[]", "log_level": "INFO", "missing_runtime_behavior": "warn", "plugin_autoupdate_last_check_duration": "10080", "raw": "false", "trusted_config_paths": "[]", "verbose": "false", "yes": "false"}

config files:
  /home/cocoonkid/.tool-versions

plugins:
  assh     https://github.com/zekker6/asdf-assh.git#f8737cc
  direnv   https://github.com/asdf-community/asdf-direnv.git#6c109a3
  go       (core)
  java     (core)
  node     (core)
  php      https://github.com/asdf-community/asdf-php.git#1eaf4de
  poetry   https://github.com/rtx-plugins/rtx-poetry.git#3d4c703
  python   (core)
  ruby     (core)
  rust     https://github.com/code-lever/asdf-rust.git#95acf4f

toolset:
  poetry@1.5.1, direnv@2.32.3, go@1.21.1, python@3.11.5, java@openjdk-21, node@20.7.0, assh@2.15.0, rust@1.72.1
```

**Comment by cocoonkid** [comment]
Update: I was able to pin it down to the java install.
All the other installs work.

![image](https://github.com/jdx/rtx/assets/43231640/3bf9e760-2dc5-483d-826d-0259ccefbabc)

**Comment by derkoe** [comment]
I see the same thing with my install - using the asdf plugin for Java works fast: https://github.com/halcyon/asdf-java:

```
rtx plugin uninstall java
rtx plugin add java https://github.com/halcyon/asdf-java.git
```

Seems like the rtx fork is broken somehow.

**Comment by derkoe** [comment]
The problem is that the rtx core plugin downloads https://joschi.github.io/java-metadata/metadata/all.json quite often. This file is nearly 36MB and GitHub pages is quite slow - so it needs around 45s for me to download.

**Comment by jdx** [comment]
what needs to happen is we need to re-host the metadata, split it up and prune information we don't need. Optionally maybe also pack it into messagepack/gzip (I'm not sure if http compression is working or not currently)

**Comment by jdx** [comment]
it seems if you want reqwest to use http compression you can't use blocking mode so that isn't going to be able to happen today, the rest of this was actually already in the metadata service though so this was a pretty simple change

**Comment by cocoonkid** [comment]
Great work @jdx!  Thanks so much!
